source/common/upstream/upstream_impl.cc

```diff
@@ -960,6 +960,44 @@ createOptions(const envoy::config::cluster::v3::Cluster& config,
       config.has_http2_protocol_options(), validation_visitor);
 }
 
+LBPolicyConfig::LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config) {
+
+  switch (config.lb_policy()) {
+    PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+  case envoy::config::cluster::v3::Cluster::ROUND_ROBIN:
+    if (config.has_round_robin_lb_config()) {
+      lbPolicy_ = std::make_unique<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>(
+          config.round_robin_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::LEAST_REQUEST:
+    if (config.has_least_request_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::LeastRequestLbConfig>(
+          config.least_request_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::RING_HASH:
+    if (config.has_ring_hash_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::RingHashLbConfig>(
+          config.ring_hash_lb_config());
+    }
+    break;
+  case envoy::config::cluster::v3::Cluster::MAGLEV:
+    if (config.has_maglev_lb_config()) {
+      lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::MaglevLbConfig>(
+          config.maglev_lb_config());
+    }
+    break;
+  default:
+    break;
+  };
+
+  if (config.has_original_dst_lb_config()) {
+    lbPolicy_ = std::make_unique<envoy::config::cluster::v3::Cluster::OriginalDstLbConfig>(
+        config.original_dst_lb_config());
+  }
```

Comment:
Should this be done before the switch (and immediately return)?
Please also add a comment of why it is not part of the switch.

Reply:
Correct me if I'm wrong but OriginalDstLB is reserved in that enum so when I try to code it, it says it's not found. But it's still apart of the oneof in the cluster proto. But I'll move the check to before the switch.